src/js/components/DataTable/DataTable.js

```diff
@@ -363,21 +403,38 @@ const DataTable = ({
               ref={bodyRef}
               cellProps={cellProps.body}
               columns={columns}
-              groupBy={groupBy.property ? groupBy.property : groupBy}
+              groupBy={typeof groupBy === 'string' ?
+                { property: groupBy} :
```

Comment:
```suggestion
                { property: groupBy } :
```
just seems like missing lint thing